EDDiscovery/EDDiscoveryForm.cs

```diff
@@ -936,7 +936,10 @@ public void StoreSystemNote(HistoryEntry he, string txt, bool send = false)
                 // Also notify various "quick" listeners
                 if (he.UpdateSystemNote(txt, send))
                 {
-                    _uncommittedNoteHistoryEntry = he;
+                    if (_uncommittedNoteHistoryEntry == null)
```

Comment:
Perhaps only set `_uncommittedNoteHistoryEntry` if `send == false`, then break the storage and sending to its own function, and call that with the `he` when `send == true`?